src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -596,9 +681,62 @@ public CreateContainerCmd fillContainerConfig(CreateContainerCmd containerConfig
             containerConfig.getHostConfig().withShmSize(shmSizeInByte);
         }
 
+        final List<String> capabilitiesToAdd = getCapabilitiesToAdd();
+        if (CollectionUtils.isNotEmpty(capabilitiesToAdd)) {
+            containerConfig.withCapAdd(toCapabilities(capabilitiesToAdd));
+        }
+
+        final List<String> capabilitiesToDrop = getCapabilitiesToDrop();
+        if (CollectionUtils.isNotEmpty(capabilitiesToDrop)) {
+            containerConfig.withCapDrop(toCapabilities(capabilitiesToDrop));
+        }
+
+        final List<String> securityOpts = getSecurityOpts();
+        if (CollectionUtils.isNotEmpty(securityOpts)) {
+            containerConfig.getHostConfig().withSecurityOpts(toSecurityOpts(securityOpts));
+        }
+
         return containerConfig;
     }
 
+    private List<Capability> toCapabilities(List<String> capabilitiesString) {
+        List<Capability> res = new ArrayList<>();
+        for(String capability : capabilitiesString) {
+            try {
+                res.add(Capability.valueOf(capability));
+            } catch(IllegalArgumentException e) {
+                throw new IllegalArgumentException("Invalid capability name : " + capability);
+            }
+        }
+        return res;
+    }
+
+    private List<String> toSecurityOpts(List<String> securityOpts) {
+        List<String> res = new ArrayList<>();
+        for (String securityOpt : securityOpts) {
+            // Seccomp : file or direct input
+            if (securityOpt.startsWith("seccomp")) {
+                String attr = securityOpt.split("=")[1];
+                Path pathAttr = Paths.get(attr);
+                if (Files.exists(pathAttr)) {
+                    try {
+                        String content = new String(Files.readAllBytes(pathAttr), StandardCharsets.UTF_8);
```

Comment:
This bit concerns me - it's reading _any_ file from the Jenkins master's filesystem, and that's going to open up a can of worms w.r.t. security.
This existing plugin code already has support for securityOpts that does not include such dangerous functionality so I'm going to keep the existing (more limited) code and not include this (IMO dangerous) code.

IMO, if we want functionality like this, it should be provided purely via the configuration WebUI and/or build pipeline code; we shouldn't be hitting the Jenkins filesystem or the docker host's filesystem directly.
i.e. users should be entering in json, not a filename of a file containing json.